sdks/python/apache_beam/io/iobase.py

```diff
@@ -1529,3 +1530,180 @@ def display_data(self):
         'source': DisplayDataItem(self.source.__class__, label='Read Source'),
         'source_dd': self.source
     }
+
+
+class SDFBoundedSourceReader(PTransform):
+  """A ``PTransform`` that uses SDF to read from each ``BoundedSource`` in a
+  PCollection.
+
+  NOTE: This transform can only be used with beam_fn_api enabled.
+  """
+  class _SDFBoundedSourceRestriction(object):
```

Comment:
This significantly duplicates functionality from `_SDFBoundedSourceWrapper` above - do you think you could have a single implementation instead of duplicating a lot of this functionality?